### ISSUE TITLE
Request to Add Support for User-Customizable Rules in Ruleset

### ISSUE THREAD

**Issue by LitDarkness** [kovzol/Java-Geometry-Expert#37]
I would like to request the ability to add user-defined rules to the ruleset, such as creating custom rules for GDD. For example, allowing the use of certain theorems to prove problems. This feature would greatly enhance the flexibility and power of the ruleset.

**Comment by kovzol** [comment]
Yes, this is already requested from other users as well. We will work on this. Meanwhile please have a look at https://github.com/opengeometryprover/GDDM/tree/master/provers/ogpgddm.
